packages/compartment-mapper/src/compartment-map.js

```diff
@@ -356,6 +356,28 @@ const assertTypes = (allegedTypes, path, url) => {
   }
 };
 
+/**
+ * @param {unknown} allegedPolicy
+ * @param {string} path
+ * @param {string} [url]
+ * @returns {asserts policy is object}
+ */
+
+export const assertPolicy = (
+  allegedPolicy,
+  path,
+  url = '<unknown-compartment-map.json>',
+) => {
+  const policy = Object(allegedPolicy);
+  assert(
+    allegedPolicy === undefined ||
+      (allegedPolicy === policy && !Array.isArray(policy)),
+    `${path}.policy must be undefined or an object, got ${allegedPolicy} in ${q(
+      url,
+    )}`,
```

Comment:
Further assertions on policy shape, while possible, might not make a lot of sense as in the future we might decouple policy implementation from hooks and make policy pluggable. That means a higher level, more approachable policy text could be possible. Stating things like "network access" or "local network access" or "same origin network access" etc.